### PR TITLE
task 3 v1

### PR DESCRIPTION
**Pull request by MaltoMight**

### CODE REVIEW

File: src/main/java/com/google/sps/data/UserProfile.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package com.google.sps.data;
+
+public final class UserProfile{
```
Comment (swissknife007) [inline src/main/java/com/google/sps/data/UserProfile.java:3]: Why are we calling this user profile?

File: src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -26,25 +29,48 @@
...
-  }
 
-  private String convertToJson(ArrayList comments) {
-    return "";
+        response.setContentType("text/html");
+        PrintWriter out = response.getWriter();
+        
+        // get all parameter names
+        Set<String> paramNames = request.getParameterMap().keySet();
+        String question = request.getParameter("question");
```
Comment (swissknife007) [inline src/main/java/com/google/sps/servlets/DataServlet.java:54]: replace strings in quotes by constant string variables please.

File: src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -26,25 +29,48 @@
...
+        String option = request.getParameter("option");
+
+        //Instantiate object for JSON
+        UserProfile profile = new UserProfile(question,option);
+        Gson gson = new Gson();
+        String json = gson.toJson(profile);
+
+
```
Comment (swissknife007) [inline src/main/java/com/google/sps/servlets/DataServlet.java:62]: please remove extra new lines.

File: src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -26,25 +29,48 @@
...
+        //Instantiate database object
+
```
Comment (swissknife007) [inline src/main/java/com/google/sps/servlets/DataServlet.java:64]: remove extra new line.

File: src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -26,25 +29,48 @@
...
+        Entity UserEntity = new Entity("User");
```
Comment (swissknife007) [inline src/main/java/com/google/sps/servlets/DataServlet.java:65]: Please rename this to something more apt.

File: src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -26,25 +29,48 @@
...
+        UserEntity.setProperty("Question", question);
```
Comment (swissknife007) [inline src/main/java/com/google/sps/servlets/DataServlet.java:66]: please use const string variables for this.

File: src/main/webapp/script.js [hunk 7]
```diff
@@ -27,8 +27,18 @@ window.onload = function () {
...
     var br = document.createElement("br");
     optionCell.appendChild(input);
     optionCell.appendChild(br);
   };
 };
+
+function getInput(){
+fetch('/data').then(response=>response.json()).then((stats) => {
+
+    console.log(stats);
```
Comment (swissknife007) [inline src/main/webapp/script.js:40]: please remove extra new lines.

File: src/main/java/com/google/sps/data/UserProfile.java (hunk 1 above)
Comment (MaltoMight) [reply src/main/java/com/google/sps/data/UserProfile.java:3]: I just noticed that I messed up with the names. I wanted use objects to convert the information into JSON by using GSON instead manually writing line by line.

File: src/main/java/com/google/sps/servlets/DataServlet.java [hunk 8]
```diff
@@ -26,25 +29,48 @@
...
+        UserEntity.setProperty("Option",option);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (oolusina) [inline src/main/java/com/google/sps/servlets/DataServlet.java:68]: Don't need to re-declare datastore object. See line 35

**Pull request merged**